cmd/xl-sets.go

```diff
@@ -1029,15 +1059,18 @@ func (s *xlSets) HealFormat(ctx context.Context, dryRun bool) (res madmin.HealRe
 		}
 	}(storageDisks)
 
-	for i, disk := range storageDisks {
-		info, err := disk.DiskInfo()
-		if err != nil {
-			storageDisks[i] = nil
-		}
-		if info.RootDisk {
-			// We should not heal on root disk. i.e in a situation where the minio-administrator has unmounted a
-			// defective drive we should not heal a path on the root disk.
-			storageDisks[i] = nil
+	infos, errs := getAllDiskInfos(storageDisks)
```

Comment:
Actually, we should avoid getAllDiskInfos() for distributed setup, i.e such that following code inside `isProductionSetup` block can be simply executed with this detection.

Reply:
Did not get you @harshavardhana

Reply:
no, getAllDisksInfo() is needed for distributed setup so that we dont heal root disks.

Reply:
as disucssed in person, this is needed.